receiver/apachereceiver/internal/configgen/component_config_gen.go

```diff
@@ -0,0 +1,27 @@
+// Code generated by mdatagen. DO NOT EDIT.
+
+package main
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/configschema"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/apachereceiver"
+)
+
+//go:generate go run ./component_config_gen.go
```

Comment:
I'm not sure if go can generate from a generated file in one shot. Likely, it means we have to run `make generate` twice. Maybe we make calling this file explicitly from `make generate` once it's generated instead of putting `go:generate` pragma here?

Reply:
No, it's recursive. That's the surprising thing I learned through this. The first generation adds the code to generate the docs, and then this code executes, all in one `make generate`.

Reply:
OK, for the record, I was wrong :/ I will look at this a bit more.